registrations/utils.py

```diff
@@ -1,6 +1,31 @@
 import phonenumbers
+from django.conf import settings
+from django.core.cache import caches
+from temba_client.v2 import TembaClient
+
+CACHE = caches["contacts"]
+CACHE_EXPIRY_SECONDS = 600
 
 
 def normalise_msisdn(msisdn):
     msisdn = phonenumbers.parse(msisdn, "ZA")
     return phonenumbers.format_number(msisdn, phonenumbers.PhoneNumberFormat.E164)
+
+
+def get_contact(msisdn):
+    contact = CACHE.get(msisdn)
+    if not contact:
+        client = TembaClient(settings.RAPIDPRO_URL, settings.RAPIDPRO_TOKEN)
```

Comment:
We probably want to declare the client outside of the function, so that the connection can be reused between requests.